src/contracts/services/derivativeScrt.ts

```diff
@@ -0,0 +1,273 @@
+  contractAddress: string,
+  codeHash?: string,
+  lcdEndpoint?: string,
+  chainId?: string,
+}) => getActiveQueryClient$(lcdEndpoint, chainId).pipe(
+  switchMap(({ client }: {client:any}) => sendSecretClientContractQuery$({
+    queryMsg: msgQueryScrtDerivativeStakingInfo(Math.round(new Date().getTime() / 1000)),
+    client,
+    contractAddress,
+    codeHash,
+  })),
+  map((response: any) => parseDerivativeScrtStakingInfo(
+    response as DerivativeScrtStakingInfoResponse,
+  )),
+  first(),
+);
+
+/**
+ * query the staking info
+ */
+async function queryDerivativeScrtStakingInfo({
+  contractAddress,
+  codeHash,
+  lcdEndpoint,
+  chainId,
+}: {
+  contractAddress: string,
+  codeHash?: string,
+  lcdEndpoint?: string,
+  chainId?: string,
+}) {
+  return lastValueFrom(queryDerivativeScrtStakingInfo$({
+    contractAddress,
+    codeHash,
+    lcdEndpoint,
+    chainId,
+  }));
+}
+
+/**
+ * parse the response for fee info
+ */
+const parseDerivativeScrtFeeInfo = (
+  response: DerivativeScrtFeeInfoResponse,
+): DerivativeScrtFeeInfo => ({
+  withdrawFee: Number(response.fee_info.withdraw) / 100000,
+  depositFee: Number(response.fee_info.deposit) / 100000,
+});
+
+/**
+ * parse the response from the batch query contract
+ */
+const parseDerivativeScrtAllInfo = (
+  response: BatchQueryParsedResponse,
+): DerivativeScrtInfo => {
+  const stakingInfoResponse = response.find(
+    (nextBatchItem: BatchQueryParsedResponseItem) => nextBatchItem.id === 'staking_info',
+  );
+  const feeInfoResponse = response.find(
+    (nextBatchItem: BatchQueryParsedResponseItem) => nextBatchItem.id === 'fee_info',
+  );
+  if (!stakingInfoResponse || !feeInfoResponse) {
+    throw new Error(`Unable to parse batch query response: ${response}`);
+  }
+  return {
+    ...parseDerivativeScrtStakingInfo(stakingInfoResponse.response),
+    ...parseDerivativeScrtFeeInfo(feeInfoResponse.response),
+  };
+};
+
+/**
+ * query the fee info
+ */
+const queryDerivativeScrtFeeInfo$ = ({
+  contractAddress,
+  codeHash,
+  lcdEndpoint,
+  chainId,
+}: {
+  contractAddress: string,
+  codeHash?: string,
+  lcdEndpoint?: string,
+  chainId?: string,
+}) => getActiveQueryClient$(lcdEndpoint, chainId).pipe(
+  switchMap(({ client }: { client: any }) => sendSecretClientContractQuery$({
+    queryMsg: msgQueryScrtDerivativeFees(),
+    client,
+    contractAddress,
+    codeHash,
+  })),
+  map((response: any) => parseDerivativeScrtFeeInfo(response as DerivativeScrtFeeInfoResponse)),
+  first(),
+);
+
+/**
+ * query the fee info
+ */
+async function queryDerivativeScrtFeeInfo({
+  contractAddress,
+  codeHash,
+  lcdEndpoint,
+  chainId,
+}: {
+  contractAddress: string,
+  codeHash?: string,
+  lcdEndpoint?: string,
+  chainId?: string,
+}) {
+  return lastValueFrom(queryDerivativeScrtFeeInfo$({
+    contractAddress,
+    codeHash,
+    lcdEndpoint,
+    chainId,
+  }));
+}
+
+/**
+ * query both the staking info and the fee info
+ */
+const queryDerivativeScrtAllInfo$ = ({
```

Comment:
we should only have this query and refer to it as queryDerivativeScrtInfo, remove the individuals.